pkg/controlplane/controller.go

```diff
@@ -89,7 +84,7 @@ type Controller struct {
 }
 
 // NewBootstrapController returns a controller for watching the core capabilities of the master
-func (c *completedConfig) NewBootstrapController(legacyRESTStorage corerest.LegacyRESTStorage, serviceClient corev1client.ServicesGetter, nsClient corev1client.NamespacesGetter, eventClient eventsv1client.EventsV1Interface, readyzClient rest.Interface) (*Controller, error) {
+func (c *completedConfig) NewBootstrapController(legacyRESTStorage corerest.LegacyRESTStorage, client kubernetes.Interface) (*Controller, error) {
```

Comment:
This makes for a simpler signature, but would also mask any expanded use of the Kubernetes API surface inside the apiserver bootstrap controllers. Is enumerating the APIs this controller touches useful? Do we want to be able to easily call arbitrary new APIs from inside these controllers?

Reply:
> ok, hopefully normal review on in-process controllers is sufficient... 

Yeah - that was my reasoning when approving this PR. We are always able to understand what new APIs are used from the code and I don't think additional code friction will change much here.

Reply:
And we don't create different clients/transports for different types